pkg/client/client.go

```diff
@@ -38,14 +40,15 @@ func NewClient(address string, certFile string, hmacKey string, logger *zap.Logg
 	// Set up the credentials for the connection.
 	perRPCHMACAuthenticator, err := auth.NewHMACAuther(logger, hmacKey, auth.EditUser)
 	if err != nil {
-		logger.Sugar().Fatalf("failed to create hmac-authenticator: %v", err)
+		return nil, errors.Wrap(err, "failed to create hmac-authenticator")
 	}
 	// TODO serverNameOverride should only be there for tests...?
 	creds, err := credentials.NewClientTLSFromFile(certFile, "metal-stack.io")
 	if err != nil {
-		logger.Sugar().Fatalf("failed to load credentials: %v", err)
+		return nil, errors.Wrap(err, "failed to load credentials")
 	}
 	opts := []grpc.DialOption{
+		grpc.WithTimeout(3 * time.Second),
```

Comment:
Needs to be addressed:
```
pkg/client/client.go:51:3: SA1019: grpc.WithTimeout is deprecated: use DialContext instead of Dial and context.WithTimeout instead.  Will be supported throughout 1.x.  (staticcheck)
		grpc.WithTimeout(3 * time.Second),
```

Reply:
This actually makes things easier, you can pass a context to the dial procedure.